.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,171 @@
+# Configuration for testing and deploying with GitHub Actions
+name: CI
+
+# Only build PRs, the master branch, and tags. Pushes to branches will only be
+# built when a PR is opened.
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
```

Comment:
Yeah, that's a bit of a compromise. Otherwise, it feels wasteful to have the CI run twice when you have a PR open (one for the PR and one for the branch). And we also encourage people to open PRs early to get the conversation started and let us know what they're working on.